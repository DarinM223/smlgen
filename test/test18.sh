../smlgen test18.sml no:s --recurmod --test
if cmp --silent test18.sml.test test18.sml.expected; then
  echo "test18 succeeded"
else
  echo "test18 failed"
  echo "Diff:"
  diff test18.sml.test test18.sml.expected
  exit 1
fi