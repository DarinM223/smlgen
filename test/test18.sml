structure Foo =
struct datatype foo = Bar of Bar.bar | Hello datatype notthis = Blah | Bleargh end

structure Bar =
struct datatype bar = Foo of Foo.foo | World datatype northis = A | B end

datatype no = No

structure A =
struct datatype a = B of int * B.a and c = C of string and d = real end

structure B = struct datatype a = B of int * A.a and c = C of string end