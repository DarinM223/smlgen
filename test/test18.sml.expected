structure Component2 =
struct
  datatype B_a =
    B_a_B of int * A_a
  and B_c =
    B_c_C of string
  and A_a =
    A_a_B of int * B_a
  and A_c =
    A_c_C of string
  and d =
    real
end
structure Component1 =
struct datatype foo = Bar of bar | Hello and bar = Foo of foo | World end
structure Foo =
struct type foo = Component1.foo datatype notthis = Blah | Bleargh end

structure Bar = struct type bar = Component1.bar datatype northis = A | B end

datatype no = No
val showNo = fn No => "No"

structure A =
struct type a = Component2.A_a type c = Component2.A_c type d = Component2.d end

structure B = struct type a = Component2.B_a type c = Component2.B_c end